### PR TITLE
Ninja jail is no longer permanent

### PR DESCRIPTION
**Pull request by Darkstick**
### Intent of your Pull Request

Once the ninja dies, the captured crew is teleported back onto the station.

This will not affect the ninja's experience, it will only mean that the captured crew will get the chance of seeing the station again if the ninja is defeated.

They will be teleported back onto the station in random but safe rooms, in a similar fashion to the laughter demon releasing their friends.

#### Changelog

:cl:  
tweak: ninja dies and crew comes back
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
laughter ninja

**Comment by Smudged** [comment]
Finally a reason to just wait and not kill myself in the old Ninja dojo.

**Comment by ynot01** [comment]
tbh i find ninja jail to be quite nice, my last experience with it we plotted methods of escape while deconstructing the entire place, making shit with autolathe, and doing botany hoping for bluespace shit

**Comment by Ktlwjec** [comment]
> tbh i find ninja jail to be quite nice, my last experience with it we plotted methods of escape while deconstructing the entire place, making shit with autolathe, and doing botany hoping for bluespace shit

Might be something for a future pr then, adding a portal that opens once the ninja is dead. So players can choose if they want to go back.

File: code/modules/antagonists/ninja/ninja.dm [hunk 1]
```diff
@@ -15,10 +15,16 @@
 
 /datum/antagonist/ninja/apply_innate_effects(mob/living/mob_override)
 	var/mob/living/M = mob_override || owner.current
+	for(var/obj/item/implant/explosive/E in M.implants)
+		if(E)
+			RegisterSignal(E, COMSIG_IMPLANT_ACTIVATED, .proc/on_death)
```
Comment (alexkar598) [inline code/modules/antagonists/ninja/ninja.dm:22]: Why?

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/ninja/ninja.dm (hunk 1 above)
Comment (SomeguyManperson) [reply code/modules/antagonists/ninja/ninja.dm:22]: issue #7175 made COMSIG_MOB_DEATH unreliable

**Comment by SomeguyManperson** [comment]
is there anything holding this up?

**Pull request merged**